amundsen_application/static/js/components/common/SearchBar/index.tsx

```diff
@@ -50,25 +59,60 @@ export class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
 
   constructor(props) {
     super(props);
+    this.refToSelf = React.createRef<HTMLDivElement>();
 
     this.state = {
+      showTypeAhead: false,
       subTextClassName: '',
       searchTerm: this.props.searchTerm,
       subText: this.props.subText,
     };
   }
 
-  static getDerivedStateFromProps(props, state) {
-    const { searchTerm } = props;
-    return { searchTerm };
+  componentWillMount = () => {
+    document.addEventListener('mousedown', this.updateTypeAhead, false);
+    /* Workaround: When returning to the HomePage, searchReset() clears the searchTerm in the
+       app state, but the change is not getting propagated. This workaround ensures the
+       searchTerm is cleared on SearchBar's state when we navigate to the HomePage */
```

Comment:
Actually I think this kind of case is what `getDerivedStateFromProps` is for, but since we were trying to move away from using that method I actually deleted it before doing the implementation to see if I could make things work without it. Let me know what you prefer -- I can put back `getDerivedStateFromProps`.

Reply:
I think this is one of the few cases that we should be using `getDerivedStateFromProps`, since we are inheriting some global state as a props and then modifying it inside the component.